python/pyarrow/tests/conftest.py

```diff
@@ -184,9 +184,16 @@ def gcs_server():
     args = [sys.executable, '-m', 'testbench', '--port', str(port)]
     proc = None
     try:
+        # check first if testbench module is available
+        import testbench  # noqa:F401
+        # start server
         proc = subprocess.Popen(args, env=env)
-    except OSError as e:
+        # Make sure the server is alive.
+        assert proc.poll() is None
+    except (ModuleNotFoundError, OSError) as e:
         pytest.skip(f"Command {args} failed to execute: {e}")
+    except AssertionError as e:
```

Comment:
I would rather have an explicit `if` rather than an assert:
```suggestion
    if proc.poll() is not None:
```